cms/src/components/images/ImageBrowser.vue

```diff
@@ -0,0 +1,71 @@
+<script setup lang="ts">
+import { db } from "@/db/baseDatabase";
+import { DocType, type ImageDto, type Uuid } from "@/types";
+import LImage from "./LImage.vue";
+import LButton from "../button/LButton.vue";
+import { PlusIcon } from "@heroicons/vue/20/solid";
+import LCard from "../common/LCard.vue";
+import ImageEditor from "./ImageEditor.vue";
+import { ref } from "vue";
+
+// TODO: Implement select event
+
+const baseUrl: string = import.meta.env.VITE_CLIENT_IMAGES_URL;
+// TODO: Replace with a generic fallback image included in project source code
+const fallbackImg: string =
+    "https://cdn.activechristianity.org/image/upload/c_thumb,w_600,g_center/v1666007697/en_ke/f7bcwvet6axzmih97pv8.jpg";
+
+const images = db.whereTypeAsRef<ImageDto[]>(DocType.Image, []);
+
+const newImage = async () => {
+    const id = db.uuid();
+    await db.upsert<ImageDto>({
+        _id: id,
+        type: DocType.Image,
+        name: "New Image",
+        description: "",
+        files: [],
+        memberOf: ["group-private-content"],
+        updatedTimeUtc: Date.now(),
+    });
```

Comment:
What do you think about abstracting this in a repository class? Or do you think that's a needless extra layer? I'm just thinking a bit about that possibly in the future there is more logic related to for example the groups, and then it might be easier to have that in a separate class.

Reply:
I am trying to use generic classes (using T extends BaseDocumentDto) which should be reusable between different document types. The base toRef function is also exposed, so if there are some queries that will not be reused, the db query function can be passed directly.

The query function calls should mostly be one-liners (need to send the initial state along, that is why it looks a little big here), so I think it should not be necessary for seperate repository files. Will see how it goes when getting to groups etc. though...